Enterprise-CMCS/macpro-mdct-qmr
# Secure Lower QMR Environments

### Description
The team has been requested to restrict lower environments to be restricted from public access and to require using cms vpn to access. 

This pull request utilizes the [cmcs waf plugin ](https://github.com/Enterprise-CMCS/serverless-waf-plugin) to apply standard firewall rules used at cms. See [this link](https://github.com/Enterprise-CMCS/serverless-waf-plugin/blob/main/src/index.ts#L73-L169) to see standard rules inherited from the plugin.

In addition to the inherited rules a new rule is applied in this PR named "vpn-only" that its default action is to restrict all access with the exception to the IP's defined in the qmr dev account ip set. 

In addition the ui serverless.yaml is where we can conditionally set if the environment is restricted. Currently it's set to a default value of restricted, with dev (master) and val also restricted and prod unrestricted per requirements of qmr being accessible to state users. 

### Related ticket(s)

---
### How to test
in a new cognito browser session visit the application endpoint for this branch both on VPN and not on VPN to confirm access or lack there of when not on VPN. 


### Important updates
development environments will not be accessible unless on VPN. 


---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
